### DISCUSSION TITLE
Regarding the Studio AI Assistant and Local LLMs

### DISCUSSION THREAD

**Discussion by jpaveg** [supabase/supabase#22014] [Feature Requests]
I was wondering if there are plans to support being able to connect to something like lmstudio, which can host a server that mimics the openai api. 

Please let me know if I'm misunderstanding this, but would reading the environment variable `OPENAI_BASE_URL` and overriding the one used in each instantiation of the openAI client (`const openai = new OpenAI({ apiKey: openAiKey, base_url: baseUrl })`) allow this to work?

I would appreciate any insight offered.

**Comment by nfairbank** [comment]
Hey, hoping to revitalize this discussion, which would be very useful. 

It would be trivial to update the openAI client as pushed out above, but it references a custom ai-commands/edge package and specifically the clippy function here:

https://github.com/supabase/supabase/blob/879a668c0c4088b20a13c46311e14624baa0ea3c/packages/ai-commands/src/docs.ts#L169

Which seems to have hardcoded openai URLs, model names, etc. Still would be relatively easy to clean up overall, but probably not as easy as changing the one-line instantiation of the client.

**Comment by mrn3** [comment]
I would love if the LLM could just run in the browser - like https://chat.webllm.ai/

After I use postgres.new for a few minutes, I get the error that I am going to melt their servers with all of my LLM requests.  I would love if I could not be limited by this.

**Comment by bobbymannino** [comment]
this would be amazing! any updates

**Comment by DoiiarX** [comment]
Wait for it

**Comment by DoiiarX** [comment]
https://github.com/orgs/supabase/discussions/33205

**Comment by YiHeCN** [comment]
If you really need this, you might want to try:

```shell
docker exec -it {YOUR_STUDIO_SERVER_CONTAINER_ID} /bin/sh
find /app -type f -exec sed -i 's/api\.openai\.com/{YOUR_OWN_API_URL}/g' {} \; 2>/dev/null
```

This is the script that changes all "[api.openai.com](http://api.openai.com/)" strings to "{YOUR_OWN_API_URL}".

It works for me, but do it at your own risk, since it might break some functionality.

<img width="659" height="214" alt="截屏2025-08-04 00 42 12" src="https://github.com/user-attachments/assets/3b514d9b-d5c1-4915-9958-981fa4d353cc" />

**Reply by YiHeCN** [reply]
Plus, if you really want to enhance it:

Run `find /app -type f -exec grep -l "api\.openai\.com" {} \; 2>/dev/null ` to locate all files containing the API URL:

```
/app/apps/studio/server.js
/app/apps/studio/.next/server/pages/api/ai/docs.js.map
/app/apps/studio/.next/server/pages/api/ai/docs.js
/app/apps/studio/.next/server/edge-chunks/656.js
/app/node_modules/.pnpm/@ai-sdk+openai@0.0.72_zod@3.23.8/node_modules/@ai-sdk/openai/dist/index.js
/app/node_modules/.pnpm/@ai-sdk+openai@0.0.72_zod@3.23.8/node_modules/@ai-sdk/openai/dist/index.mjs
/app/node_modules/.pnpm/openai@4.71.1_encoding@0.1.13_zod@3.23.8/node_modules/openai/index.js
/app/node_modules/.pnpm/openai@4.71.1_encoding@0.1.13_zod@3.23.8/node_modules/openai/index.mjs
```


One of these files may contain the most effective configuration to modify.

**Reply by YiHeCN** [reply]
Additionally, it uses gpt-4o-mini by default, so if you're running with an OpenAI-compatible model, you can repeat the process and change the "gpt-4o-mini" string to "claude-3.5-sonnet" or whatever model you prefer.

```bash
find /app -type f -exec sed -i 's/gpt-4o-mini/{YOUR_OPENAI_COMPATIBLE_MODEL}/g' {} \; 2>/dev/null
```

Before and after change:
<img width="652" height="340" alt="截屏2025-08-04 01 01 05" src="https://github.com/user-attachments/assets/7cce2d2f-00e5-45bc-840e-9b6fa88ffbe8" />

**Reply by kimsoo527** [reply]
@nfairbank

**Reply by DoiiarX** [reply]
> Additionally, it uses gpt-4o-mini by default, so if you're running with an OpenAI-compatible model, you can repeat the process and change the "gpt-4o-mini" string to "claude-3.5-sonnet" or whatever model you prefer.
> 
> ```shell
> find /app -type f -exec sed -i 's/gpt-4o-mini/{YOUR_OPENAI_COMPATIBLE_MODEL}/g' {} \; 2>/dev/null
> ```
> 
> Before and after change: <img alt="截屏2025-08-04 01 01 05" width="652" height="340" src="https://private-user-images.githubusercontent.com/102681555/473800616-7cce2d2f-00e5-45bc-840e-9b6fa88ffbe8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQzMDg3ODksIm5iZiI6MTc1NDMwODQ4OSwicGF0aCI6Ii8xMDI2ODE1NTUvNDczODAwNjE2LTdjY2UyZDJmLTAwZTUtNDViYy04NDBlLTliNmZhODhmZmJlOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgwNFQxMTU0NDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ZGNiZDdlOGNjMDE4M2YxMzBkZGUwOTdlZTQ2NDYzYzI1NTU3Zjc2MjYzZTlkNTg5ZGU0Yjc3M2Y2NTJiMTY4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.dbURpqWqQ1G22-HFUf8pFG6pEK3uZsvpVzkSiiBy7t8">

大哥。佩服。

**Comment by andresvidal** [comment]
Any updates on this? Why can't this be set in `.env` files like: 

**With Groq's API**
OPENAI_API_BASE=https://api.groq.com/openai/v1
OPENAI_API_KEY=gsk_...
OPENAI_MODEL=openai/gpt-oss-120b
